app/models/job_execution.rb

```diff
@@ -168,6 +168,7 @@ def setup(dir)
   end
 
   def resolve_ref_to_commit
+    @repository.update_mirror
```

Comment:
hmm this should happen somewhere in the setup ? ... kinda surprising if it happens here ?

Reply:
As explained in PR comments, this calls `commit_from_ref` which calls `ensure_mirror_current` which doesn't do what it says it does.